software/spmd/interrupt_tests/dual_threading_test.S

```diff
@@ -0,0 +1,117 @@
+// this program creates a tiny threading package. The threads communicate with each other using remote interruts.
```

Comment:
@taylor-bsg is this what you were referring to in the threading test? The vanilla core is able to switch between the two threads and exit the handler correctly to the alternate thread. I have enabled trace for both of them and I can see that the PCs are being traced correctly for each thread.

Reply:
What did you exactly mean by the following?

> Perhaps you could code that up as a test, and combine with the trace
> interrupt to see if you can trace one thread and not the other and get the
> exact right sequence of PC’s?

Reply:
This is good. But maybe have some kind of verification on the trace interrupt itself? Perhaps have it increment a counter based on which thread it was tracing in, which you check against the correct answer?